pkg/ingester/ingester_test.go

```diff
@@ -6607,3 +6663,46 @@ func testIngesterCanEnableIngestAndQueryNativeHistograms(t *testing.T, sampleHis
 
 	testResult(expectedMatrix, "Result should contain the histogram even when not accepting histograms")
 }
+
+func TestIngester_GetOpenTSDBsConcurrencyConfig(t *testing.T) {
+	tests := map[string]struct {
+		walReplayConcurrency               int
+		maxTSDBOpeningConcurrencyOnStartup int
+		tenantCount                        int
+		expectedTSDBOpenConcurrency        int
+		expectedTSDBWALReplayConcurrency   int
+	}{
+		"if -blocks-storage.tsdb.wal-replay-concurrency is 0, -blocks-storage.tsdb.max-tsdb-opening-concurrency-on-startup is used": {
+			walReplayConcurrency:               0,
+			maxTSDBOpeningConcurrencyOnStartup: 10,
+			tenantCount:                        5,
+			expectedTSDBOpenConcurrency:        10,
+			expectedTSDBWALReplayConcurrency:   0,
+		},
+		"if -blocks-storage.tsdb.wal-replay-concurrency > 0 and there are <= 10 tenants, ignore -blocks-storage.tsdb.max-tsdb-opening-concurrency-on-startup and parallelize WAL replay on sequential openings": {
+			walReplayConcurrency:               3,
+			maxTSDBOpeningConcurrencyOnStartup: 10,
+			tenantCount:                        5,
+			expectedTSDBOpenConcurrency:        1,
+			expectedTSDBWALReplayConcurrency:   3,
+		},
+		"if -blocks-storage.tsdb.wal-replay-concurrency > 0 and there are > 10 tenants, ignore -blocks-storage.tsdb.max-tsdb-opening-concurrency-on-startup and parallelize openings with single WAL replay": {
+			walReplayConcurrency:               3,
+			maxTSDBOpeningConcurrencyOnStartup: 10,
+			tenantCount:                        5,
+			expectedTSDBOpenConcurrency:        1,
+			expectedTSDBWALReplayConcurrency:   3,
```

Comment:
This test is identical than the previous one, while it should test the case there are more than 10 tenants.

```suggestion
			tenantCount:                        15,
			expectedTSDBOpenConcurrency:        3,
			expectedTSDBWALReplayConcurrency:   1,
```

Reply:
True, too much Unit tests written today :)